content/en/docs/concepts/workloads/controllers/daemonset.md

```diff
@@ -171,9 +168,8 @@ the related features.
 | `node.kubernetes.io/unreachable`         | NoExecute  | `TaintBasedEvictions`                                        | 1.8+    | When `TaintBasedEvictions` is enabled, they will not be evicted when there are node problems such as a network partition. |
 | `node.kubernetes.io/disk-pressure`       | NoSchedule |                                                              | 1.8+    |                                                              |
 | `node.kubernetes.io/memory-pressure`     | NoSchedule |                                                              | 1.8+    |                                                              |
-| `node.kubernetes.io/unschedulable`       | NoSchedule | `ScheduleDaemonSetPods`                                      | 1.11+   | When `ScheduleDaemonSetPods` is enabled, DaemonSet pods tolerate unschedulable attributes by default scheduler. |
-| `node.kubernetes.io/network-unavailable` | NoSchedule | `ScheduleDaemonSetPods` (pods using hostNetwork)             | 1.11+   | When `ScheduleDaemonSetPods` is enabled, DaemonSet pods, who uses host network, tolerate network-unavailable attributes by default scheduler. |
-| `node.kubernetes.io/out-of-disk`         | NoSchedule | `ExperimentalCriticalPodAnnotation` (critical pod only)      | 1.8+    |                                                              |
+| `node.kubernetes.io/unschedulable`       | NoSchedule |                                                              | 1.12+   | DaemonSet pods tolerate unschedulable attributes by default scheduler.                                                    |
```

Comment:
@k82cn Hmm... I didn't see any PR mentioning that.

Also I did a test: create a two-node cluster, then go to node2 and disconnect with apiserver (using iptables command), then node2 is expected to have taint applied:

```yaml
spec:
  taints:
  - effect: NoSchedule
    key: node.kubernetes.io/unreachable
    timeAdded: 2018-09-12T17:54:52Z
```

But when I create a deamonset, only node1 is landed with a pod:

```
NAME          READY     STATUS    RESTARTS   AGE       IP           NODE          NOMINATED NODE
pause-rwp5n   1/1       Running   0          7m20s     10.244.2.4   kube-node-1   <none>
```

And here is the toleration list of daemonset pod on node1:

```yaml
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/disk-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/memory-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/unschedulable
    operator: Exists
```

Did I miss anything?

Reply:
@Huang-Wei doesn't your experiment verify that `not-ready` and `unreachable` tolerations are added as well?

Reply:
@bsalamat that's `NoExecute` effect. I think @k82cn meant `NoSchedule` effect.

Reply:
Here is a tolerations list when running ds in v1.11.0:

```yaml
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/disk-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/memory-pressure
    operator: Exists
```

 I think the behavior in 1.12 is good - the only additional toleration is `node.kubernetes.io/unschedulable`. That's as expected because we're explicitly handling unschedulable case by leveraging taint/tolerations mechanism (`kubectl taint node` will add an explicit taint for that node).

Reply:
oh, really sorry, confused with `TaintNodesByCondition` :(
I'm ok with the doc.